component/notification/src/main/java/org/exoplatform/social/notification/plugin/SocialNotificationUtils.java

```
@@ -352,10 +350,16 @@ public static String buildRedirecActivityUrl(String type, String id, String acti
   public static void addFooterAndFirstName(String remoteId, TemplateContext templateContext) {
     try {
       Identity receiver = Utils.getIdentityManager().getOrCreateIdentity(OrganizationIdentityProvider.NAME, remoteId, true);
-      templateContext.put("FIRSTNAME", (String) receiver.getProfile().getProperty(Profile.FIRST_NAME));
-      templateContext.put("FOOTER_LINK", LinkProviderUtils.getRedirectUrl("notification_settings", receiver.getRemoteId()));
+      String firstName = "";
+      String redirectUrl = "";
+      if(receiver != null) {
+        firstName = (String) receiver.getProfile().getProperty(Profile.FIRST_NAME);
+        redirectUrl = LinkProviderUtils.getRedirectUrl("notification_settings", receiver.getRemoteId());
+      }
+      templateContext.put("FIRSTNAME", firstName);
+      templateContext.put("FOOTER_LINK", redirectUrl);
     } catch (Exception e) {
-      return;
+      LOG.error("Error while getting firstname and notification settings link of user " + remoteId, e);
```

If we want to continue to process the rest of content, we should add empty strings for:
+      templateContext.put("FIRSTNAME", "");
+      templateContext.put("FOOTER_LINK", "");

Else IMO, it could be better to throw the exception to let know the top layer that there was an exception and let it display and handle it, WDYT ?

---

> If we want to continue to process the rest of content, we should add empty strings for:

The initial purpose was to handle the normal behavior of activities posted by application (not users). If we set empty strings in case of others unexpected errors, it means that notifications could be sent with the wrong data.

> Else IMO, it could be better to throw the exception to let know the top layer that there was an exception and let it display and handle it, WDYT ?

ok